static/app/utils/replays/hooks/useReplayList.tsx

```diff
@@ -0,0 +1,171 @@
+import {useCallback, useEffect, useState} from 'react';
+import * as Sentry from '@sentry/react';
+import type {Location} from 'history';
+import {stringify} from 'query-string';
+
+import type {Organization} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {getLocalToSystem, getUserTimezone, getUtcDateString} from 'sentry/utils/dates';
+import EventView from 'sentry/utils/discover/eventView';
+import {decodeInteger} from 'sentry/utils/queryString';
+import {mapResponseToReplayRecord} from 'sentry/utils/replays/replayDataUtils';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useApi from 'sentry/utils/useApi';
+import {useLocation} from 'sentry/utils/useLocation';
+import type {ReplayListLocationQuery, ReplayListRecord} from 'sentry/views/replays/types';
+
+const DEFAULT_LIMIT = 50;
+export const DEFAULT_SORT = '-startedAt';
+
+export const REPLAY_LIST_FIELDS = [
+  'countErrors',
+  'duration',
+  'finishedAt',
+  'id',
+  'projectId',
+  'startedAt',
+  'urls',
+  'user',
+];
+
+type Options = {
+  eventView: EventView;
+  organization: Organization;
+  defaultLimit?: number;
+};
+
+type State = {
+  fetchError: undefined | RequestError;
+  isFetching: boolean;
+  pageLinks: null | string;
+  replays: undefined | ReplayListRecord[];
+};
+
+type Result = State;
+
+function useReplayList({
+  eventView,
+  organization,
+  defaultLimit = DEFAULT_LIMIT,
+}: Options): Result {
+  const api = useApi();
+  const location = useLocation<ReplayListLocationQuery>();
+
+  const [data, setData] = useState<State>({
+    fetchError: undefined,
+    isFetching: true,
+    pageLinks: null,
+    replays: [],
+  });
```

Comment:
Do we have a hook for this data fetching?

Reply:
not sure what you mean in here.